File: exercism/python/proverb/proverb.py
"""
proverb
"""


def proverb(*inputs, qualifier=None):
    """
    Given a list of inputs, generate the relevant proverb
    """

    if not inputs:
        return []

    data = [f"For want of a {inputs[index]} the {inputs[index +1]} was lost."
            for index, _ in enumerate(inputs[:-1])]

    data.append(f"And all for the want of a {qualifier + ' ' if qualifier else ''}{inputs[0]}.")

    return data

File: exercism/go/proverb/proverb.go
package proverb

import "fmt"

// Given a list of inputs, generate the relevant proverb
func Proverb(rhyme []string) []string {
	data := []string{}
	if len(rhyme) > 0 {
		for index := 0; index < len(rhyme)-1; index++ {
			data = append(data, fmt.Sprintf("For want of a %s the %s was lost.", rhyme[index], rhyme[index+1]))
		}
		data = append(data, fmt.Sprintf("And all for the want of a %s.", rhyme[0]))
	}

	return data
}
